keras_nlp/samplers/sampler.py

```diff
@@ -31,15 +31,16 @@
         cache: Optional. A tensor or nested structure of tensors that will be
             updated by each call to `next`. This can be used to cache
             computations from early iterations of the generative loop.
-        index: Optional. The first index to start sampling at.
+        index: Optional. The first index of `prompt` to start sampling at.
+            Usually this is set as the length of the shortest non-padding
+            sequence in `prompt`.
         mask: Optional. A 2D integer tensor with the same shape as `prompt`.
             Locations which are `True` in the mask are never updated during
-            sampling. Often this will mark all ids in `prompt` which were
-            present in the original input.
+            sampling. Often non-padding tokens in `prompt` are `True` in `mask`.
```

Comment:
This sentence feels unclear still, maybe...

Usually used to mark all locations in the dense `prompt` tensor which were present in a user input.